### PR TITLE
node: skip docker ping check when using fake docker

### PR DESCRIPTION
**Pull request by csrwng**
When using fake docker
- Skips ping check 
- Sets kubelet cgroup driver to default (cgroupfs)

### CODE REVIEW

**Comment by csrwng** [comment]
@deads2k we need to be able to run the kubelet with a fake docker client (which is doable when you set `DOCKER_HOST=fake://`). However, we implement our own version of Ping() which doesn't use the fake client. This change makes it so we tolerate a failure to Ping.

/cc @stevekuznetsov @smarterclayton

**Comment by stevekuznetsov** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17979/openshift/origin/integration/10

**Comment by stevekuznetsov** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17979/openshift/origin/integration/11

**Comment by stevekuznetsov** [comment]
Just the one failure now:

```
runner_test.go:124: FAILED TestDiagNodeConditions, retrying:

diag_nodes_test.go:42: expected no warnings with one node ready, but: []types.DiagnosticError{types.DiagnosticError{ID:"DClu0002", LogMessage:"Node 127.0.0.1 is defined but is not marked as ready.\nReady status is False because \"KubeletNotReady\"\nIf the node is not intentionally disabled, check that the master can\nreach the node hostname for a health check and the node is checking in\nto the master with the same hostname.\n\nWhile in this state, pods should not be scheduled to deploy on the node,\nand any existing scheduled pods will be considered failed and removed.\n", Cause:error(nil)}}
etcd.go:140: dumping etcd to "/tmp/openshift/test-integration/tmp-testdiagnodeconditions138363114/etcd-dump-TestDiagNodeConditions-v2.json"
```

File: pkg/cmd/server/kubernetes/node/node.go [hunk 1]
```diff
@@ -112,9 +112,18 @@ func (c *NodeConfig) EnsureDocker(docker *dockerutil.Helper) {
...
-		return
+	_, isFakeDocker := client.(*dockertools.FakeDockerClient)
+	if isFakeDocker {
+		// If using the fake docker client, ensure that the CgroupDriver for the kubelet matches
+		// the default cgroup driver.
+		c.KubeletServer.CgroupDriver = "cgroupfs"
+	}
+
+	if !isFakeDocker {
+		if err := dockerClient.Ping(); err != nil {
```
Comment (soltysh) [inline pkg/cmd/server/kubernetes/node/node.go]: Why not adding the `Ping` method to the `FakeDockerClient` rather than introducing this hack?

**Comment by deads2k** [comment]
> @deads2k we need to be able to run the kubelet with a fake docker client (which is doable when you set DOCKER_HOST=fake://). However, we implement our own version of Ping() which doesn't use the fake client. This change makes it so we skip the Ping when using the fake Docker client
> 
> /cc @stevekuznetsov @smarterclayton

I'm ok with it.  It's fairly likely to slip during a rebase (type assertions break all the time), so you may want a test.

**Comment by stevekuznetsov** [comment]
As soon as this is passing our containerized tests it can't slip in a rebase since it will be required to run integration tests in CI

File: pkg/cmd/server/kubernetes/node/node.go (hunk 1 above)
Comment (csrwng) [reply pkg/cmd/server/kubernetes/node/node.go]: We'd still need to know whether we're using it so we can match the cgroupdriver

Comment (csrwng) [reply pkg/cmd/server/kubernetes/node/node.go]: I will open a PR upstream

**Comment by deads2k** [comment]
> As soon as this is passing our containerized tests it can't slip in a rebase since it will be required to run integration tests in CI

@csrwng if that's the case, consider yourself on notice for rebase help if this borks badly :)

/lgtm

**Comment by csrwng** [comment]
This is needed to unblock integration tests. I will create a follow up issue to clean it up

File: pkg/cmd/server/kubernetes/node/node.go (hunk 1 above)
Comment (soltysh) [reply pkg/cmd/server/kubernetes/node/node.go]: Thanks.

**Pull request merged**